node/core/dispute-coordinator/src/lib.rs

```diff
@@ -393,11 +401,20 @@ async fn get_rolling_session_window<Context>(
 
 /// Wait for `ActiveLeavesUpdate`, returns `None` if `Conclude` signal came first.
 #[overseer::contextbounds(DisputeCoordinator, prefix = self::overseer)]
-async fn wait_for_first_leaf<Context>(ctx: &mut Context) -> Result<Option<ActivatedLeaf>> {
+async fn wait_for_first_leaf<Context>(
+	ctx: &mut Context,
+	sync_oracle: &Box<dyn SyncOracle + Send + Sync>,
+) -> Result<Option<ActivatedLeaf>> {
 	loop {
 		match ctx.recv().await? {
 			FromOrchestra::Signal(OverseerSignal::Conclude) => return Ok(None),
 			FromOrchestra::Signal(OverseerSignal::ActiveLeaves(update)) => {
+				if sync_oracle.is_major_syncing() {
```

Comment:
I feel like this could be racy:

1. We receive the signal while still syncing, but we catch up quickly afterwards.
2. `is_major_syncing` returns false, although we were syncing when receiving that update.

Reply:
TL;DR: This PR is old and I have forgotten to close it in time. 

Yes, it is. I was relying on the fact that we don't get `ActiveLeaves` during major sync (unless there is a reorg) so this should be the initial leaf. 

But then I quickly realized this is a bad assumption and thought about putting a flag in `ActiveLeavesUpdate` indicating if we are in initial major sync.

Then @ordian filed https://github.com/paritytech/polkadot/issues/6694 and we decided to fix the problem more generally by blocking signals in overseer during initial major sync.

The result is https://github.com/paritytech/polkadot/pull/6689 which is still work in progress.